GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -524,7 +524,12 @@ private void OnNewFileCreated(string relativePath, bool isDirectory)
                         }
                         else
                         {
-                            this.FileSystemCallbacks.OnFolderCreated(relativePath);
+                            if (this.FileSystemCallbacks.OnFolderCreated(relativePath))
+                            {
+                                // Need to enumerate the directory to get and create placeholders

```

Comment:
Could you update this comment to include what `OnFileCreated` returning true means? (I saw you mentioned that in the Windows callback code).

Reply:
I think it is hard to understand from just reading the callsite what the return value indicates (none of the other callback methods have a return value).  As an alternative, maybe have an outparam that returns a bool, that way you can document how the value is to be interpreted in the method signature.

Reply:
I like that idea @jamill Thanks!